### PR TITLE
Fix invalid series titles in visualization tooltips

### LINKED ISSUE #11395: Column "Custom title" not working in tooltips
**Describe the bug**
1.
if char has many lines(visualization), when user try to custom display name;
the display name does not effect all lines;
if you change first display name, it just effect line1, like this;
label1 -> line1
label2 -> line2
...
labelN  -> lineN 

2. data x-axis is readonly and how to set display name? 

**Screenshots**

**Information about your Metabase Installation:**

You can get this information by going to Admin -> Troubleshooting.

- Chrome 78.0.3904.97
- Windows 10
- Metabase version v0.33.5.1
- Metabase hosting environment: Jar-file on centos
- Metabase internal database: MySQL

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### LINKED ISSUE #15150: Wrong series label in multiple series scatterplot
**Describe the bug**
I have a PostgreSQL table with one categorical column and four numeric columns. I made a scatterplot with one of the numeric columns as the X axis and the other 3 numerics as separate series.

When hovering over a data point, the pop-up sometimes uses the wrong label. For example, in this screenshot, the series label FNR is shown twice. One the two FNR labels corresponds to a value from the actual FNR series, and the other FNR label corresponds to a value from a different series.

Screen Shot 2021-03-12 at 11 26 59 AM

The actual SQL column names are correct and the chart legend is also correct. The labels get confused when hovering on some series, but not others.

This may be related to older closed bugs:
https://github.com/metabase/metabase/issues/3669
https://github.com/metabase/metabase/issues/3599

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_1_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.82 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.10+9",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.10",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.10+9",
    "os.name": "Linux",
    "os.version": "4.14.209-160.339.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.6"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.18"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2021-03-03",
      "tag": "v0.38.1",
      "branch": "release-x.38.x",
      "hash": "79ef63a"
    },
    "settings": {
      "report-timezone": "US/Eastern"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by alxnddr**
### Description

1) On scatter plots with two or more series were used wrong `seriesTitle` to show on the tooltip, it was hardcoded to take the first series title. That is why tooltips had a duplication of labels but values were correct.
2) Series names manually set in settings were not used to show labels inside the tooltip.

Fixes  https://github.com/metabase/metabase/issues/15150
Fixes https://github.com/metabase/metabase/issues/11395

### CODE REVIEW

**Comment by nemanjaglumac** [comment]
I updated the label from `.Reproduced` to `support/Repro`. We'll apply `.Reproduced` to the related issue once this is merged.

File: frontend/test/metabase/scenarios/visualizations/scatter.cy.spec.js
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/visualizations/scatter.cy.spec.js:19]: Just a word of caution - normal user is not allowed to do some (more sensitive) requests. For now, this works but you may find yourself in a situation where you need more complicated preparation for some test. In that case, you'd need to use `cy.signInAsAdmin()` - which is what we use most of the time.

For some specific areas (permissions, sandboxing, etc.) it is extremely important to run the same test against many different user roles. But even then admin creates the setup for the test, and then we can sign is as a desired user role for the assertion phase.

No need to change anything here, btw. I thought this is just a good place to tell you about this :)

File: frontend/test/metabase/scenarios/visualizations/scatter.cy.spec.js
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/visualizations/scatter.cy.spec.js:24]: if we have this in both tests, it's completely ok to extract it to top-most `beforeEach`.

**Comment by nemanjaglumac** [comment]
Tests look good. 🎉
Let's wait on someone to approve FE code.

File: frontend/src/metabase/visualizations/lib/apply_tooltips.js
Comment (paulrosenzweig) [inline frontend/src/metabase/visualizations/lib/apply_tooltips.js:39]: This could be `getIn(settings, ["series_settings", col.name, "title"])` to save some lines.

File: frontend/src/metabase/visualizations/lib/apply_tooltips.js
Comment (paulrosenzweig) [inline frontend/src/metabase/visualizations/lib/apply_tooltips.js:26]: I'm not sure why this was set to `0` before. I wondered if it was due to some data only being set on the first card in a breakout multi-series, but I didn't see anything to confirm that. Anyways, the new code makes more sense!

**Review by paulrosenzweig — APPROVED** [review]
LGTM! Nicely done!

File: frontend/test/metabase/scenarios/visualizations/scatter.cy.spec.js
Comment (alxnddr) [reply frontend/test/metabase/scenarios/visualizations/scatter.cy.spec.js:19]: Good to know, thanks for sharing

File: frontend/src/metabase/visualizations/lib/apply_tooltips.js
Comment (alxnddr) [reply frontend/src/metabase/visualizations/lib/apply_tooltips.js:27]: Yes, I was not sure too. It caused showing wrong series name in the tooltip as described in this issue https://github.com/metabase/metabase/issues/15150

Thanks for the review 🙏

**Pull request merged**
